From 539f80d531dea2e3ec9501becd949f344201f0e9 Mon Sep 17 00:00:00 2001
From: Scott Stephens <stephens.js@gmail.com>
Date: Thu, 7 Aug 2025 12:55:26 -0400
Subject: [PATCH] Restore environment when exiting FAPIConfig context

---
 src/tpm2_pytss/FAPI.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/tpm2_pytss/FAPI.py b/src/tpm2_pytss/FAPI.py
index df665c1..9809b0b 100644
--- a/src/tpm2_pytss/FAPI.py
+++ b/src/tpm2_pytss/FAPI.py
@@ -110,7 +110,10 @@ class FAPIConfig(contextlib.ExitStack):
     def __exit__(self, exc_type, exc_val, exc_tb):
         super().__exit__(exc_type, exc_val, exc_tb)
 
-        del os.environ[FAPI_CONFIG_ENV]
+        if self.config_env_backup is not None:
+            os.environ[FAPI_CONFIG_ENV] = self.config_env_backup
+        else:
+            del os.environ[FAPI_CONFIG_ENV]
 
         if self.config_tmp_path is not None:
             os.unlink(self.config_tmp_path)
-- 
2.50.1

